tests/foreman/cli/test_product.py

```diff
@@ -321,33 +257,35 @@ def test_positive_update_4(self, prod_name):
         @Assert: Product Renamed to original
 
         """
-        prod = make_product({
-            u'name': prod_name,
-            u'organization-id': self.org['id'],
-        })
-        new_prod_name = gen_string('alpha', 8)
-        # Update the product name
-        Product.update({
-            u'id': prod['id'],
-            u'name': new_prod_name,
-        })
-        # Verify Updated
-        prod = Product.info({
-            u'id': prod['id'],
-            u'organization-id': self.org['id'],
-        })
-        self.assertEqual(prod['name'], new_prod_name)
-        # Now, Rename product to original
-        Product.update({
-            u'id': prod['id'],
-            u'name': prod_name,
-        })
-        prod = Product.info({
-            u'id': prod['id'],
-            u'organization-id': self.org['id'],
-        })
-        # Verify renamed back to Original name.
-        self.assertEqual(prod['name'], prod_name)
+        for prod_name in generate_strings_list():
+            with self.subTest(prod_name):
+                prod = make_product({
+                    u'name': prod_name,
+                    u'organization-id': self.org['id'],
+                })
```

Comment:
I thought about it before updating this test and decided to go with creating the product everytime because this test does some complicated steps of creating the product, updating it and again reverting the name :)